protobuf/v1/snapshot.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+import "nexus.proto";
+package mayastor.v1;
+
+service SnapshotRpc {
+    // Common snapshot related methods.
+    rpc CreateReplicaSnapshot(CreateReplicaSnapshotRequest) returns (CreateReplicaSnapshotResponse) {}
+    rpc CreateNexusSnapshot(NexusCreateSnapshotRequest) returns (NexusCreateSnapshotResponse) {}
+    rpc ListSnapshot(ListSnapshotsRequest) returns (ListSnapshotsResponse) {}
+    rpc DeleteSnapshot(DeleteSnapshotRequest) returns (google.protobuf.Empty) {}
+}
+
+/// Parameter details of a single snapshot
+message SnapshotInfo {
+    string snapshot_uuid = 1; // uuid of the snapshot
+    string snapshot_name = 2; // name of the snapshot
+    uint64 snapshot_size = 3; // amount of bytes referenced by snapshot
+    uint64 num_clones = 4; // number of clones created from this snapshot
+    google.protobuf.Timestamp timestamp = 5; // snapshot creation_timestamp
+    string source_uuid = 6; // uuid of the source from which snapshot is created (nexus / replica)
+    uint64 source_size = 7; // amount of bytes referenced by source
+    string entity_id = 8; // uuid of the entity for which snapshot is taken.
+    string txn_id = 9; // unique transaction id for snapshot.
+    bool valid_snapshot = 10; // true: valid, false: invalid (missing one/more snapshotdescriptor field i.e. txn_id, entity_id, source_uuid
+}
+/// Request parameters to list snapshot.
+message ListSnapshotsRequest {
+    optional string source_uuid = 1;  // source uuid from which snapshot is taken
+}
+
+/// List Snapshot Response.
+message ListSnapshotsResponse {
+    repeated SnapshotInfo snapshots = 1; // list of Snapshot Params.
+}
+/// Delete Snapshot Request.
+message DeleteSnapshotRequest {
+    string snapshot_uuid = 1; // snapshot to be deleted
+    oneof pool {
+        string pool_name = 2; // name of the pool where the snapshot resides.
```

Comment:
I think this means we now should also store this information in the snapshot itself?

Reply:
> I think this means we now should also store this information in the snapshot itself?

You mean to save it in control plane side right ? From dataplane point of view it is just the validation. But from CP point of view, node and pool information mapping to snapshot can be saved.

Reply:
I mean, on the data returned from the grpc list_snapshot, dataplane could include this pool name or id.
So when we're doing a list all snapshots we have the information ready? Otherwise we then have to list all replicas and iteratively find the replica that matches the snapshot and gets its pool.

Reply:
got it. I think `pool_uuid` can be more approprite to return as part of list_snapshot return parameter.
Same can be return as part of CreateSnapshot return parameter as well.